activerecord/lib/active_record/persistence.rb

```diff
@@ -477,11 +477,21 @@ def touch(*names, time: nil)
           changes[column] = write_attribute(column, time)
         end
 
-        changes[self.class.locking_column] = increment_lock if locking_enabled?
-
         clear_attribute_changes(changes.keys)
         primary_key = self.class.primary_key
-        self.class.unscoped.where(primary_key => self[primary_key]).update_all(changes) == 1
+        scope = self.class.unscoped.where(primary_key => send(primary_key))
```

Comment:
If we have to get what is the primary_key anyway I think we could just get the value of the attribute without having to ask again what is the primary key